source/expressions.tex

```diff
@@ -2610,45 +2610,51 @@
 semantic properties proceed in the following order:
 
 \begin{itemize}
-\item Substitution of template arguments (if any)
-into the \grammarterm{expression} is performed.
-
-\item If the \tcode{noexcept} specifier is present,
-\tcode{E} shall not be a potentially-throwing expression\iref{except.spec}.
-
-\item If the \grammarterm{return-type-requirement} is present, then:
-
-\begin{itemize}
-\item Substitution of template arguments (if any)
-into the \grammarterm{return-type-requirement} is performed.
-
-\item If the \grammarterm{return-type-requirement} is a
-\grammarterm{trailing-return-type},
-%%% FIXME: is -> shall be
-\tcode{E} is implicitly convertible to
-the type named by the \grammarterm{trailing-return-type}.
-If conversion fails, the enclosing \grammarterm{requires-expression}
-is \tcode{false}.
-
-\item If the \grammarterm{return-type-requirement}
-starts with a \grammarterm{constrained-parameter}\iref{temp.param},
-the \grammarterm{expression} is deduced against
-an invented function template \tcode{F}
-using the rules in \ref{temp.deduct.call}.
-\tcode{F} is a \tcode{void} function template
-with a single type template parameter \tcode{T}
-declared with the \grammarterm{constrained-parameter}.
-A \grammarterm{cv-qualifier-seq} \cv{} is formed
-as the union of \tcode{const} and \tcode{volatile} specifiers
-around the \grammarterm{constrained-parameter}.
-\tcode{F} has a single parameter
-whose \grammarterm{type-specifier} is \cv{}~\tcode{T}
-followed by the \grammarterm{abstract-declarator}.
-%%% FIXME: Remove this; if deduction fails, the construct should
-%%% be ill-formed.
-If deduction fails,
-the enclosing \grammarterm{requires-expression} is \tcode{false}.
-\end{itemize}
+\item
+  Substitution of template arguments (if any)
+  into the \grammarterm{expression} is performed.
+\item
+  If the \tcode{noexcept} specifier is present,
+  \tcode{E} shall not be a potentially-throwing expression\iref{except.spec}.
+\item
+  If the \grammarterm{return-type-requirement} is present, then:
+  \begin{itemize}
+  \item
+    Substitution of template arguments (if any)
+    into the \grammarterm{return-type-requirement} is performed.
+  \item
+    If the \grammarterm{return-type-requirement} is a
+    \grammarterm{trailing-return-type}\iref{dcl.decl},
+    %%% FIXME: is -> shall be
+    \tcode{E} is implicitly convertible to
+    the type named by the \grammarterm{trailing-return-type}.
+    If conversion fails, the enclosing \grammarterm{requires-expression}
+    is \tcode{false}.
+  \item
+    If the \grammarterm{return-type-requirement} is
```

Comment:
If the _return-type-requirement_ has a [ ... ].

Reply:
thanks!